float64[] pathx
float64[] pathy
float64 w
---

